packages/relay/src/lib/eth.ts

```diff
@@ -1696,6 +1696,20 @@ export class EthImpl implements Eth {
    */
   async getTransactionByHash(hash: string, requestIdPrefix?: string) {
     this.logger.trace(`${requestIdPrefix} getTransactionByHash(hash=${hash})`, hash);
+
+    // check if tx is synthetic and exists in cache
+    const cacheKeySyntheticLog = `${constants.CACHE_KEY.SYNTHETIC_LOG_TRANSACTION_HASH}${hash}`;
+    const cachedLog = await this.cacheService.getSharedWithFallback(
```

Comment:
Q: should we guard this on `this.shouldPopulateSyntheticContractResults` rather so that the cache query is only made when applicable?

Reply:
agreed

Reply:
✅  done